### PR TITLE
cockroachdb: no-longer rely on auto-init

### PR DESCRIPTION
**Pull request by irfansharif**
When starting CockroachDB nodes, always include join flags. In order to
actually initialize the cluster, rely on an explicit `cockroach init`.

Previously any node started without explicit join flags was tasked with
bootstrapping the cluster. This was deprecated behavior, and was removed
in https://github.com/cockroachdb/cockroach/pull/51245.

Touches https://github.com/cockroachdb/cockroach/issues/49360
Touches https://github.com/cockroachdb/cockroach/issues/51052
Fixes https://github.com/cockroachdb/cockroach/issues/51050
Fixes https://github.com/cockroachdb/cockroach/issues/51725
Fixes https://github.com/cockroachdb/cockroach/issues/51726
Fixes https://github.com/cockroachdb/cockroach/issues/51727
Fixes https://github.com/cockroachdb/cockroach/issues/51728
Fixes https://github.com/cockroachdb/cockroach/issues/51729
Fixes https://github.com/cockroachdb/cockroach/issues/51730
Fixes https://github.com/cockroachdb/cockroach/issues/51731
Fixes https://github.com/cockroachdb/cockroach/issues/51732
Fixes https://github.com/cockroachdb/cockroach/issues/51733
Fixes https://github.com/cockroachdb/cockroach/issues/51734
Fixes https://github.com/cockroachdb/cockroach/issues/51735
Fixes https://github.com/cockroachdb/cockroach/issues/51736
Fixes https://github.com/cockroachdb/cockroach/issues/51737
Fixes https://github.com/cockroachdb/cockroach/issues/51740
Fixes https://github.com/cockroachdb/cockroach/issues/51741
Fixes https://github.com/cockroachdb/cockroach/issues/51742
Fixes https://github.com/cockroachdb/cockroach/issues/51743
Fixes https://github.com/cockroachdb/cockroach/issues/51744
Fixes https://github.com/cockroachdb/cockroach/issues/51745
Fixes https://github.com/cockroachdb/cockroach/issues/51746
Fixes https://github.com/cockroachdb/cockroach/issues/51747
Fixes https://github.com/cockroachdb/cockroach/issues/51748
Fixes https://github.com/cockroachdb/cockroach/issues/51749
Fixes https://github.com/cockroachdb/cockroach/issues/51750
Fixes https://github.com/cockroachdb/cockroach/issues/51751
Fixes https://github.com/cockroachdb/cockroach/issues/51752
Fixes https://github.com/cockroachdb/cockroach/issues/51753
Fixes https://github.com/cockroachdb/cockroach/issues/51754
Fixes https://github.com/cockroachdb/cockroach/issues/51755
Fixes https://github.com/cockroachdb/cockroach/issues/51756
Fixes https://github.com/cockroachdb/cockroach/issues/51758

### CODE REVIEW

**Comment by knz** [comment]
Beware that two of the linked issues are real issues currently being investigated. Please ensure that this PR does not close these legitimate issues. (If you've already taken care of that please disregard)
-- 
Verstuurd vanaf mijn Android apparaat met K-9 Mail. Excuseer mijn beknoptheid.

File: cockroachdb/src/jepsen/cockroach.clj
Comment (knz) [inline cockroachdb/src/jepsen/cockroach.clj:71]: I don't think this wait is necessary any more. It used to be necessary back in the day where it was unsafe to start a client while the node wasnt fully started yet. We fixed it.

File: cockroachdb/src/jepsen/cockroach.clj
Comment (knz) [inline cockroachdb/src/jepsen/cockroach.clj:75]: ditto

Comment (knz) [reply cockroachdb/src/jepsen/cockroach.clj:75]: Also the code below "restart cluster to work around balancing bug" is not necessary any more. 

(I would rather update this code to instead wait for up-replication, if there's an easy way to add that)

**Review by knz — APPROVED** [review]

**Comment by irfansharif** [comment]
TFTR! Like discussed on slack I'm thinking to hold off on removing the sleeps+restart, just to keep this diff focused on the more recent breakage. I'll wait for @bdarnell as well here before merging.

> Please ensure that this PR does not close these legitimate issues. (If you've already taken care of that please disregard)

I did miss that, done.

File: cockroachdb/src/jepsen/cockroach.clj
Comment (bdarnell) [reply cockroachdb/src/jepsen/cockroach.clj:75]: How long have these things been fixed? Remember that we run jepsen (from the same branch) against all supported release branches of cockroachdb. 

Even if the sleeps aren't necessary to work around old bugs, we shouldn't completely get rid of them until we're able to add an explicit "wait for upreplication/rebalancing" step.

**Review by bdarnell — APPROVED** [review]

File: cockroachdb/src/jepsen/cockroach.clj
Comment (knz) [reply cockroachdb/src/jepsen/cockroach.clj:75]: Ah good point. Maybe keep them then.

**Pull request merged**
